### PR TITLE
Introduce logstash [+ plus restructuring]

### PR DESCRIPTION
**Pull request by evilmarty**

### CODE REVIEW

**Comment by evilmarty** [comment]
Build is failing because I've been using version 8.0 of Terraform while I think the build agents run 7.0.* 😞

File: elasticsearch/vpc.tf [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+resource "aws_security_group_rule" "elb_http" {
+  type        = "ingress"
+  from_port   = 80
+  to_port     = 80
+  protocol    = "tcp"
+  cidr_blocks = ["${data.aws_vpc.selected.cidr_block}"]
+  security_group_id = "${aws_security_group.elb.id}"
+}
+
+resource "aws_security_group_rule" "elb_https" {
```
Comment (elliot) [inline elasticsearch/vpc.tf]: The ELB doesn't have a listener on this port

File: lambda.tf [hunk 2]
```diff
@@ -0,0 +1,55 @@
+module "curator" {
+  source = "github.com/everydayhero/npm-lambda-packer"
+
+  package = "logar-curator"
+  version = "1.0.1"
+
+  environment = <<ENVIRONMENT
```
Comment (elliot) [inline lambda.tf:7]: Lambda environment can now be [specified](https://www.terraform.io/docs/providers/aws/r/lambda_function.html#environment) in the resource. Useful in future

File: main.tf [hunk 3]
```diff
@@ -4,226 +4,81 @@ provider "aws" {
...
-  vpc_id = "${var.vpc_id}"
-
-  egress {
-    from_port = 80
-    to_port   = 80
-    protocol  = "tcp"
-    security_groups = ["${module.elasticsearch.security_group_id}"]
+  filter {
+    name = "name"
+    values = ["*/hvm-ssd/ubuntu-xenial-16.04-amd64-server*"]
```
Comment (elliot) [inline main.tf:13]: This should be looking for EDH hardened images

File: elasticsearch/user-data.txt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  data: /srv/elasticsearch/data
+discovery:
+  type: ec2
+  zen.minimum_master_nodes: ${minimum_master_nodes}
+  ec2.groups: "${security_groups}"
+cloud.aws.region: "$AWS_REGION"
+EOF
+
+# Must be changed when bootstrap.mlockall: true
+sed -ie "s%^#LimitMEMLOCK=infinity$%LimitMEMLOCK=infinity%" /usr/lib/systemd/system/elasticsearch.service
```
Comment (elliot) [inline elasticsearch/user-data.txt:62]: Should also set the ES_HEAP_SIZE here

https://www.elastic.co/guide/en/elasticsearch/guide/current/heap-sizing.html

File: lambda.tf (hunk 2 above)
Comment (evilmarty) [reply lambda.tf:7]: I know but the focus of this PR is to upgrade ES and use Logstash

File: main.tf (hunk 3 above)
Comment (evilmarty) [reply main.tf:13]: No it shouldn't. It's a generic AMI finder while we specify the actual AMI we want to use during build time.

File: elasticsearch/vpc.tf (hunk 1 above)
Comment (evilmarty) [reply elasticsearch/vpc.tf]: Removed

File: elasticsearch/user-data.txt (hunk 4 above)
Comment (evilmarty) [reply elasticsearch/user-data.txt:62]: Updated

**Pull request merged**